pytorch_forecasting/metrics/tests/test_all_metrics.py

```diff
@@ -0,0 +1,458 @@
+            prediction = metric_instance.to_prediction(
+                y_pred,
+            )
+        except Exception as e:
+            pytest.skip(
+                f"to_prediction failed with {test_case} for {metric_instance.__class__.__name__}: {e}"  # noqa: E501
+            )
+
+        assert isinstance(prediction, torch.Tensor)
+
+        if y_pred.ndim == 3:
+            assert prediction.ndim == 2
+            assert prediction.shape[:2] == y_pred.shape[:2]
+        # else:
+        #     assert prediction.shape == y_pred.shape
+
+    @pytest.mark.parametrize("test_case", DATA_FORMAT_NAMES)
+    def test_to_quantiles_method(self, metric_instance, test_data, test_case):
+        """Test the ``to_quantiles`` method of metrics."""
+
+        if metric_instance is None:
+            pytest.skip("No valid metric instance provided.")
+
+        try:
+            test_case_data = test_data[test_case]
+            y_pred = test_case_data["y_pred"]
+            quantiles = [0.1, 0.5, 0.9]
+            if metric_instance.__class__.__name__ == "QuantileLoss":
+                quantile_pred = metric_instance.to_quantiles(y_pred)
+            else:
+                quantile_pred = metric_instance.to_quantiles(
+                    y_pred, quantiles=quantiles
+                )  # noqa: E501
+        except Exception as e:
+            pytest.skip(
+                f"to_quantiles failed with {test_case} for {metric_instance.__class__.__name__}: {e}"  # noqa: E501
+            )
+        assert isinstance(quantile_pred, torch.Tensor)
+
+        if metric_instance.__class__.__name__ == "QuantileLoss":
+            assert quantile_pred.shape == y_pred.shape
+        else:
+            assert quantile_pred.ndim == 3
+            if y_pred.ndim == 2 and metric_instance.__class__.__name__ != "PoissonLoss":  # noqa: E501
+                assert quantile_pred.shape[2] == 1
+            else:
+                assert quantile_pred.shape[2] == len(quantiles)
+
+    def test_metric_compositions(self, metric_class):
+        """Test correct working of metric compositions via CompositeMetric"""
+
+        if not issubclass(metric_class, Metric):
+            pytest.skip(
+                f"{metric_class.__name__} is not a Metric subclass,",
+                " skipping test.",
+            )
+
+        metric_a = metric_class()
+        metric_b = metric_class()
+
+        composite = metric_a + metric_b
+        assert isinstance(composite, CompositeMetric)
+
+        weighted = metric_a * 0.5
+        assert isinstance(weighted, CompositeMetric)
+
+        composite_weighted = metric_a + metric_b * 0.5
+        assert isinstance(composite_weighted, CompositeMetric)
+
+    def test_doctest_examples(self, metric_class):
+        """Run doctests for metric class."""
+        try:
+            from skbase.utils.doctest_run import run_doctest
+
+            run_doctest(metric_class, name=f"class {metric_class.__name__}")
+        except ImportError:
```

Comment:
always available when tests run - since we are installing the `tests` depset which also contains `pytest`